File: frontend/src/types/question.ts
// Question types mirrored from backend src/schemas/question.py
// CRITICAL: QuestionListItem must NOT have answer_en/answer_pt/explanation_en/explanation_pt

import type { Difficulty } from "./ui_meta";

export interface QuestionListItem {
  id: number;
  theme_id: number;
  prompt_en: string;
  prompt_pt: string;
  tricky: boolean;
  tricky_pattern: string | null;
  difficulty: Difficulty;
  translations_status: string;
  image_url?: string | null;
}

export interface QuestionDetail extends QuestionListItem {
  answer_en: string;
  answer_pt: string;
  explanation_en: string;
  explanation_pt: string;
}

File: backend/src/schemas/question.py
"""Question schemas."""



from pydantic import BaseModel

from .ui_meta import Difficulty


class QuestionListItem(BaseModel):
    """Question summary for list/m(mock-test) UIs — NO answers or explanations."""

    model_config = {"from_attributes": True}

    id: int
    theme_id: int
    prompt_en: str
    prompt_pt: str
    tricky: bool
    tricky_pattern: str | None = None
    difficulty: Difficulty
    translations_status: str  # e.g. "complete", "en_only", "pt_only"
    image_url: str | None = None


class QuestionDetail(QuestionListItem):
    """Full question with answers and explanations — only for admin/review UIs."""

    model_config = {"from_attributes": True}

    answer_en: str
    answer_pt: str
    explanation_en: str
    explanation_pt: str
    image_url: str | None = None  # inherited from QuestionListItem, explicit for clarity
